Darwin/Services/StitchRemoteMongoDBService/StitchSyncPerformanceTests/SyncPerformanceIntTests.swift

```diff
@@ -36,82 +56,36 @@ class SyncPerformanceIntTests: XCTestCase {
-            print("PerfLog: Test: \(numDoc) docs of size \(docSize)")
-            try assertEqual(Int.self, ctx.coll.count([:]) ?? 1, 0)
-
-            var docs = getDocuments(numDocs: numDoc, docSize: docSize)
-            ctx.coll.insertMany(docs, joiner.capture())
-            let _: Any? = try joiner.value()
-            try assertEqual(Int.self, ctx.coll.count([:]) ?? 0, numDoc)
-
-            ctx.coll.sync.configure(conflictHandler: DefaultConflictHandler<Document>.remoteWins())
-
-            docs = getDocuments(numDocs: numDoc, docSize: docSize)
-            ctx.coll.sync.insertMany(&docs)
-            _ = try ctx.coll.sync.proxy.dataSynchronizer.doSyncPass()
-            try assertEqual(Int.self, ctx.coll.sync.syncedIds().count, numDoc)
-
-            docs = getDocuments(numDocs: numDoc, docSize: docSize)
-            ctx.coll.sync.insertMany(&docs)
-            _ = try ctx.coll.sync.proxy.dataSynchronizer.doSyncPass()
-            try assertEqual(Int.self, ctx.coll.sync.syncedIds().count, 2 * numDoc)
-
-            try assertEqual(Int.self, ctx.coll.count([:]) ?? 0, 3 * numDoc)
-        })
-    }
-
-    func getDocuments(numDocs: Int, docSize: Int) -> [Document] {
-        return (0..<numDocs).map {iter in
-            guard let newDoc = try? ["_id": ObjectId(),
-                                     "data": Binary(data: Data(repeating: UInt8(iter % 100), count: docSize),
-                                                    subtype: Binary.Subtype.userDefined)] as Document else {
-                fatalError("Failed to create \(numDocs) documents of size \(docSize)")
+        harness.runPerformanceTestWithParameters(
+            testName: "intialSyncLocal",
+            runId: runId,
+            testDefinition: {ctx, numDoc, docSize in
+                print("PerfLog: Test: \(numDoc) docs of size \(docSize)")
+
+                if numDoc > 0 {
+                    let docs = SyncPerformanceTestUtils.generateDocuments(numDoc: numDoc, docSize: docSize)
+                    ctx.coll.insertMany(docs, joiner.capture())
+                    let _: Any? = try joiner.value()
+                    try assertEqual(Int.self, ctx.coll.count([:]) ?? 0, numDoc)
+                }
+
+                let count = ctx.coll.count([:])
+                try assertEqual(Int.self, count ?? 0, numDoc)
+
+                ctx.coll.sync.configure(conflictHandler: DefaultConflictHandler<Document>.remoteWins())
+
+                if numDoc > 0 {
+                    var docs = SyncPerformanceTestUtils.generateDocuments(numDoc: numDoc, docSize: docSize)
+                    ctx.coll.sync.insertMany(&docs)
+                    _ = try ctx.coll.sync.proxy.dataSynchronizer.doSyncPass()
+                    try assertEqual(Int.self, ctx.coll.sync.syncedIds().count, numDoc)
+                }
+            }, beforeEach: {_, numDoc, docSize in
+                print("PerfLog: (Custom Setup) \(numDoc) docs of size \(docSize)")
```

Comment:
[nit] we should have a private log method that accepts some kind of tag to dedupe the code here and have consistent logging